subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -467,21 +467,60 @@ static int is_abort_cb(void *next, void *curr, lll_prepare_cb_t *resume_cb)
 
 	/* Different radio event overlap */
 	if (next != curr) {
+		struct lll_scan_aux *lll_aux;
 		struct lll_scan *lll;
 
 		lll = ull_scan_lll_is_valid_get(next);
-		if (!lll) {
-			struct lll_scan_aux *lll_aux;
-
-			lll_aux = ull_scan_aux_lll_is_valid_get(next);
-			if (IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC_SKIP_ON_SCAN_AUX) ||
-			    !lll_aux) {
-				/* Abort current event as next event is not a
-				 * scan and not a scan aux event.
-				 */
-				return -ECANCELED;
+		if (lll) {
+			/* Do not abort current periodic sync event as next
+			 * event is a scan event.
+			 */
+			return 0;
+		}
+
+		lll_aux = ull_scan_aux_lll_is_valid_get(next);
+		if (!IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC_SKIP_ON_SCAN_AUX) &&
+		    lll_aux) {
+			/* Do not abort current periodic sync event as next
+			 * event is a scan aux event.
+			 */
+			return 0;
+		}
+
+#if defined(CONFIG_BT_CTLR_SCAN_AUX_SYNC_RESERVE_MIN)
+		struct lll_sync *lll_sync_next;
+		struct lll_sync *lll_sync_curr;
+
+		lll_sync_next = ull_sync_lll_is_valid_get(next);
+		if (!lll_sync_next) {
+			/* Abort current event as next event is not a
+			 * scan and not a scan aux event.
+			 */
+			return -ECANCELED;
+		}
+
+		lll_sync_curr = curr;
+		if (lll_sync_curr->abort_count < lll_sync_next->abort_count) {
+			if (lll_sync_curr->abort_count < UINT8_MAX) {
+				lll_sync_curr->abort_count++;
 			}
+
+			/* Abort current event as next event has higher abort
+			 * count.
+			 */
+			return -ECANCELED;
+		}
```

Comment:
Do we ever risk both `lll_sync_curr->abort_count` and `lll_sync_next->abort_count` to become `UINT8_MAX`? In which case, neither of them can increment, and the events will never get aborted? 
```
lll_sync_curr->abort_count < lll_sync_next->abort_count

Reply:
By implementation a max of 250 periodic sync are supported  (limited by ticker's use of uint8_t for ids, and 5 other reserved tickers).

the `abort_count` (max at 249) will not reach UINT8_MAX.

Say both does reach UINT8_MAX, then `return 0` is returned here:
https://github.com/zephyrproject-rtos/zephyr/pull/49579/commits/3259230635fe357755a8ee53eae61d48fdf1fd8b#diff-3b747a1cde6e2938850148bd938b27771b7a9cc2be571792ebf3d5ddfbeb5918R526-R529

current event (that occurs first in timeline) will always continue and next event will always be aborted (eventually leading to sync loss). As UINT8_MAX will not be reached, there is no issue.